exercises/basic/basic.p4

```diff
@@ -70,15 +85,42 @@ control MyVerifyChecksum(inout headers hdr, inout metadata meta) {
 
 /*************************************************************************
 **************  I N G R E S S   P R O C E S S I N G   *******************
+* High-level intent:
+*   - Do an LPM lookup on IPv4 dstAddr
+*   - On hit, call ipv4_forward(next-hop MAC, output port)
+*   - Otherwise, drop or NoAction (as configured)                         *
 *************************************************************************/
 
 control MyIngress(inout headers hdr,
                   inout metadata meta,
                   inout standard_metadata_t standard_metadata) {
+
     action drop() {
         mark_to_drop(standard_metadata);
     }
 
+    /*********************************************************************
+     * NOTE FOR NEW READERS:
+     * 'ipv4_forward(dstAddr, port)' is invoked by table 'ipv4_lpm'.
+     *
+     * The values for 'dstAddr' and 'port' are *action data* supplied by
+     * the control plane when it installs entries in 'ipv4_lpm'.
+     *
+     * They mean:
+     *   - dstAddr  => Ethernet destination MAC for the next hop
+     *   - port     => output port (ultimately written to standard_metadata.egress_spec)
+     *
+     * Example (BMv2 simple_switch_CLI):
+     *   table_add ipv4_lpm ipv4_forward 10.0.1.1/32 => 00:00:00:00:01:00 1
+     * which passes MAC=00:00:00:00:01:00 and PORT=1 as action parameters
+     * into ipv4_forward(dstAddr, port).
+     *
+     * The exercise keeps the actual logic as a TODO so learners implement it:
```

Comment:
You have nearly identical comments before ipv4_forward, and also within ipv4_forward.  I'd recommend keeping one of them and removing the other.  I don't have a strong preference for which.